### ISSUE TITLE
What are bounding box labels for?

### ISSUE THREAD

**Issue by colinator** [yformer/EfficientSAM#14]
In this code:

```
def run_ours_box(img_path, pts_sampled, model):
    bbox = torch.reshape(torch.tensor(pts_sampled), [1, 1, 2, 2])
    bbox_labels = torch.reshape(torch.tensor([2, 3]), [1, 1, 2])
    image = cv2.imread(image_path)
    image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)

    predicted_logits, predicted_iou = model(
        img_tensor[None, ...].cuda(),
        bbox.cuda(),
        bbox_labels.cuda(),
    )
```
... what are the bbox_labels doing? Why are they [2,3], and why does the model need them? I couldn't find that described in the paper either. Maybe I missed it...

thx!

**Comment by balakv504** [comment]
The model uses the same interface to encode both points and bounding boxes. The bbox_labels (in this case) that is to tell the model that the input points are corners of the bounding box (rather than just points inside the sought mask). Bounding box corner gets a label of 2 and 3 respectively. Points within a mask gets a label of 1.

**Comment by colinator** [comment]
Ah, check, thx.
